app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,74 @@
+class CategoriesController < ApplicationController
+  before_action :set_category, only: %i[show edit update destroy]
+  before_action :authenticate_user!
+
+  # GET /categories or /categories.json
+  def index
+    @categories = Category.where(user_id: current_user.id)
+  end
+
+  # GET /categories/1 or /categories/1.json
+  def show
+    @category = Category.find(params[:id])
+    @category_transactions = @category.transactions.all
+  end
+
+  # GET /categories/new
+  def new
+    @category = Category.new
+  end
+
+  # GET /categories/1/edit
+  def edit; end
+
+  # POST /categories or /categories.json
+  def create
+    # @category = Category.new(category_params)
+    # @category.user_id = current_user.id
```

Comment:
Please remove commented code from here as it can make the whole project unclean.